plugin/src/main/java/hudson/plugins/swarm/PluginImpl.java

```diff
@@ -221,6 +227,38 @@ public void doCreateSlave(StaplerRequest req, StaplerResponse rsp, @QueryParamet
         }
     }
 
+    /**
+     * Allows the client to mark a slave offline
+     */
+    public void doMarkSlaveOffline(StaplerRequest req, StaplerResponse rsp, @QueryParameter String name,
+                              @QueryParameter String reason, @QueryParameter String secret) throws IOException {
+        if (!getSwarmSecret().equals(secret)) {
+            rsp.setStatus(SC_FORBIDDEN);
+            return;
+        }
+
+        Jenkins jenkins = Jenkins.getInstance();
+
+        jenkins.checkPermission(SlaveComputer.DISCONNECT);
+
+        Node node = getNodeByName(name, rsp);
+
+        node.toComputer().setTemporarilyOffline(true, new OfflineCause.UserCause(User.current(), reason));
+
+        normalResponse(req, rsp, "<response/>"); //TODO: not sure what to send here
```

Comment:
I'm not sure either offhand. Assuming that the default response status is `javax.servlet.http.HttpServletResponse.SC_OK`, can we rely on the response status alone and simply avoid calling `rsp.getCompressedWriter()` in the first place for such responses? I can look into this further if you're having trouble.